sklearn/model_selection/tests/test_split.py

```diff
@@ -1571,3 +1572,82 @@ def test_leave_p_out_empty_trainset():
             ValueError,
             match='p=2 must be strictly less than the number of samples=2'):
         next(cv.split(X, y, groups=[1, 2]))
+
+
+def test_binnedstratifiedkfold_error():
+    with pytest.raises(ValueError, match='Need at least two bins'):
+        BinnedStratifiedKFold(n_bins=1)
+
+
+def test_binnedstratifiedkfold_balance():
+    rng = np.random.RandomState(0)
+    for _ in range(10):
+        n_splits = 2 + int(10 * rng.rand())
+        X, y = make_regression(noise=10, random_state=rng)
+        train_sizes = []
+        test_sizes = []
+
+        bskf = BinnedStratifiedKFold(n_splits=n_splits)
+
+        for train_index, test_index in bskf.split(X=X, y=y):
+            train_sizes.append(len(train_index))
+            test_sizes.append(len(test_index))
+
+        assert (np.max(test_sizes) - np.min(test_sizes)) <= 1
+        assert (np.max(train_sizes) - np.min(train_sizes)) <= 1
+        assert np.sum(test_sizes) == X.shape[0]
+        assert np.sum(train_sizes) == X.shape[0] * (n_splits - 1)
+
+
+def test_binnedstratifiedkfold_stable_moments_between_splits():
+    """check if BinnedStratifiedKFold performs on average better than KFold in
+    terms of lower between-fold variance of fold mean(y_test) and fold
+    std(y_test)
+    """
+    binned_has_more_stable_std_list = []
+    binned_has_more_stable_mean_list = []
+    rng = np.random.RandomState(0)
+
+    for trial in range(100):
+        n_splits = 2 + int(10*rng.rand())
+        X, y = make_regression(noise=10, random_state=rng)
+
+        ymeans_binned = []
+        ystds_binned = []
+
+        skf = BinnedStratifiedKFold(n_splits=n_splits)
+
+        kf = KFold(n_splits=n_splits, shuffle=True, random_state=42)
```

Comment:
By not shuffling the skf are you disadvantaging it (which is good for the test)? If so add a comment?